bin/dfx-sns-propose

```diff
@@ -10,38 +10,11 @@ source "$SOURCE_DIR/clap.bash"
 clap.define short=i long=identity desc="The dfx identity to use" variable=DFX_IDENTITY default="$(dfx identity whoami)"
 clap.define short=n long=network desc="The dfx network to use" variable=DFX_NETWORK default="local"
 clap.define short=N long=neuron desc="The dfx network to use" variable=DFX_NEURON_ID default=""
+clap.define long=save-proposal-id-to desc="Passed to sns propose --save-to" variable=PROPOSAL_ID_FILE default=""
 # Source the output file ----------------------------------------------------------
 source "$(clap.build)"
 
 DFX_NEURON_ID="${DFX_NEURON_ID:-$(dfx-neuron-id --identity "$DFX_IDENTITY" --network "$DFX_NETWORK")}"
-DFX_IDENTITY_PEM="$(dfx-identity-pem --identity "$DFX_IDENTITY")"
 DFX_NNS_URL="$(dfx-network-provider --network "$DFX_NETWORK")"
-export DFX_NEURON_ID DFX_IDENTITY_PEM DFX_NNS_URL
 
-# The due date of the sale must be greater than 1 day at the time of NNS proposal execution
-SWAP_DUE_TIMESTAMP="$(perl -e 'print time() + 14 * 24 * 3600')"
-SNS_SWAP="$(dfx canister id sns_swap --network "$DFX_NETWORK")"
-SNS_TOKENS_E8s=1000314100000000
-TOKEN_SYMBOL="$(awk '/token_symbol:/{print $2}' sns.yml || echo SOMETHING)"
-PROPOSAL_TITLE="Proposal to create an SNS-DAO for $TOKEN_SYMBOL"
-set ic-admin \
-  --secret-key-pem "$DFX_IDENTITY_PEM" \
-  --nns-url "$DFX_NNS_URL" \
-  propose-to-open-sns-token-swap \
-  --proposer "$DFX_NEURON_ID" \
-  --min-participants 1 \
-  --min-icp-e8s 5000000000 \
-  --max-icp-e8s 314100000000 \
-  --min-participant-icp-e8s 10000000 \
-  --max-participant-icp-e8s 314100000000 \
-  --swap-due-timestamp-seconds "$SWAP_DUE_TIMESTAMP" \
-  --sns-token-e8s "$SNS_TOKENS_E8s" \
-  --target-swap-canister-id "$SNS_SWAP" \
-  --neuron-basket-count 2 \
-  --neuron-basket-dissolve-delay-interval-seconds 2629800 \
-  --proposal-title "$PROPOSAL_TITLE" \
-  --summary "Some summary"
-
-printf ' "%q"' "${@}"
-echo
-"${@}"
+bin-other/sns propose --neuron-id "$DFX_NEURON_ID" ${PROPOSAL_ID_FILE:+--save-to "$PROPOSAL_ID_FILE"} --network "$DFX_NNS_URL" sns.yml
```

Comment:
which step creates `sns.yml`? the `dfx-sns-config-random`?

Reply:
Correct.
On line 52:
```
sed "$SED_COMMANDS" "$SNS_INIT_PATH" >sns.yml
```
`>sns.yml` sends the output of the command to that file.

It's also documented in the help text of `dfx-sns-config-random`:
```
Creates an sns.yml to configure an SNS swap.
```